application/src/objects/obstacles/abstract-dynamic-obstacle.js

```diff
@@ -0,0 +1,51 @@
+'use strict';
+
+var Phaser = require('phaser');
+
+var AbstractDynamicObstacle = function(state, x, y, key, collisionGroup)
+{
+    Phaser.Sprite.apply(this, [state.game, x, y, key]);
+
+    this.createPhysicsBody(state);
+
+    this.body.setCollisionGroup(collisionGroup);
```

Comment:
I guess I'm not quite sure either. Since I think our calls to `body.collides()` should be made within  the state (since the state will be keeping track of all collision groups if we have more than one), then perhaps it makes sense to let this get called from the state object as well.